webapp/channels/src/components/root/actions.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import type {History} from 'history';
+
+import type {UserProfile} from '@mattermost/types/users';
+
+import {getFirstAdminSetupComplete} from 'mattermost-redux/actions/general';
+import {getProfiles} from 'mattermost-redux/actions/users';
+import {General} from 'mattermost-redux/constants';
+import {getIsOnboardingFlowEnabled} from 'mattermost-redux/selectors/entities/preferences';
+import {getActiveTeamsList} from 'mattermost-redux/selectors/entities/teams';
+import {checkIsFirstAdmin, getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import type {ThunkActionFunc} from 'mattermost-redux/types/actions';
+
+import * as GlobalActions from 'actions/global_actions';
+
+import {StoragePrefixes} from 'utils/constants';
+
+export function redirectToOnboardingOrDefaultTeam(history: History): ThunkActionFunc<void> {
```

Comment:
Since there's only a couple of these, do we want to put them in the same file as some of the other actions? Or are we thinking of making more "component-specific" actions files? Personally I'd rather try and group together the actions as best we can, but only 1/5 really.

Reply:
I've been preferring more component-specific actions for a while since it helps to keep related code together rather than dividing it up by the type of code that it is. I think it also helps to normalize getting as much non-rendering logic out of the component as possible because the alternative is to pass more data from Redux into the component which would slow down rendering